BitwardenShared/UI/Platform/Application/AppProcessor.swift

```
@@ -183,8 +183,7 @@ public class AppProcessor {
         )
 
         await services.migrationService.performMigrations()
-        await services.environmentService.loadURLsForActiveAccount()
-        _ = await services.configService.getConfig()
+        await prepareEnvironmentConfig()
```

🤔 If there's an app upgrade, and you open the extension using one of the non-interactive flows prior to running the app, we'd skip performing any migrations here, which could be problematic. Would it be better to pass a flag to `AppProcessor.start()` indicating if user interaction is available as opposed to breaking out the environment config pieces? It's maybe more branching, but might be better for maintainability long-term to ensure `start()` is always called.

---

I think that if migrations are needed I would throw an error in the flow and ask the user to open the main app to perform such migrations. I feel it's safer to run them in the main app process given the restrictions we have in the autofill extension process and that it may run in the background where several things are omitted/break.
Would you agree? If so, I'll create a ♻️ task for it and we can handle that in a future PR.